third_party/move/move-vm/runtime/src/loader/script.rs

```diff
@@ -43,18 +40,13 @@ pub struct Script {
 impl Script {
     pub(crate) fn new(
         script: Arc<CompiledScript>,
-        struct_ty_storage: &impl StructTypeStorage,
         struct_name_index_map: &StructNameIndexMap,
     ) -> PartialVMResult<Self> {
         let mut struct_names = vec![];
         for struct_handle in script.struct_handles() {
             let struct_name = script.identifier_at(struct_handle.name);
             let module_handle = script.module_handle_at(struct_handle.module);
             let module_id = script.module_id_for_handle(module_handle);
-
-            struct_ty_storage
```

Comment:
same question as for modules / struct_ty_storage / compatibility checks

Reply:
For compat checks: I removed it to avoid more calls to module storage and because they were duplicated (unlike paranoid mode, duplicated I think). `dependencies::verify_module` does this check based on dependencies, and we construct the module immediately after, so they bring no value?

I chatted to @runtian-zhou about this before, and also had a post on slack ....

Reply:
Sweet, this is great, missed the context but good that I noticed given the missed context xD